clients/java/src/main/java/io/camunda/zeebe/client/impl/response/DeploymentEventImpl.java

```diff
@@ -15,21 +15,42 @@
  */
 package io.camunda.zeebe.client.impl.response;
 
+import io.camunda.zeebe.client.api.response.Decision;
+import io.camunda.zeebe.client.api.response.DecisionRequirements;
 import io.camunda.zeebe.client.api.response.DeploymentEvent;
 import io.camunda.zeebe.client.api.response.Process;
-import io.camunda.zeebe.gateway.protocol.GatewayOuterClass.DeployProcessResponse;
+import io.camunda.zeebe.gateway.protocol.GatewayOuterClass.DeployResourceResponse;
+import io.camunda.zeebe.gateway.protocol.GatewayOuterClass.Deployment;
 import java.util.List;
 import java.util.stream.Collectors;
 
 public final class DeploymentEventImpl implements DeploymentEvent {
 
   private final long key;
   private final List<Process> processes;
+  private final List<Decision> decisions;
+  private final List<DecisionRequirements> decisionRequirements;
 
-  public DeploymentEventImpl(final DeployProcessResponse response) {
+  public DeploymentEventImpl(final DeployResourceResponse response) {
     key = response.getKey();
     processes =
-        response.getProcessesList().stream().map(ProcessImpl::new).collect(Collectors.toList());
+        response.getDeploymentsList().stream()
```

Comment:
:wrench: We iterate 3 times over the same list, creating three different lists. I understand these lists are expected to be small, so it's not that big of a deal, but it's probably not a good habit to take, as I'm not sure this is much more readable than, say:

```java
    for (final Deployment deployment : response.getDeploymentsList()) {
      switch (deployment.getMetadataCase()) {
        case PROCESS:
          processes.add(deployment.getProcess());
          break;
        case DECISION:
          decisions.add(deployment.getProcess());
          break;
        case DECISIONREQUIREMENTS:
          decisionRequirements.add(deployment.getProcess());
          break;
        case METADATA_NOT_SET:
          // maybe explode? idk
          break;
      }
    }
```

I thought about using `Collectors.groupingBy(Deployment::getMetadataCase)`, which works, but the collected lists are of type Deployment, so not super useful, you'd still need to iterate over them and cast them.

Note that the above has the nice case of checking for a deployment with an unknown metadata case, which could be useful to discover errors.

Reply:
I have to read a bit about what `METADATA_NOT_SET` actually means, to determine what we should do there.

What do you think about logging in the default case? So in case it has new deployment metadata that is not yet known to this client because its outdated. We could then inform the user to update the client to a newer version. Does that make sense? And if so, should we log that at info level?